src/react/utils.js

```diff
@@ -566,6 +566,46 @@ export function evaluateWithNestedEffects(
   }
 }
 
+// create a new effects object with the bindings, property bindings
+// and created objects of both effects combined into a single effects
+// also ensure use the value and generator from the target effect are
+// used in the new effects object
+function createMergedEffects(realm: Realm, target: Effects, source: Effects): Effects {
+  let [sc, pg, pb, pp, po] = target;
+  let [, , sb, sp, so] = source;
+  let result = construct_empty_effects(realm);
+  let [, , rb, rp, ro] = result;
+
+  result[0] = sc;
+  result[1] = pg;
+
+  if (pb) {
+    pb.forEach((val, key, m) => rb.set(key, val));
+  }
+  sb.forEach((val, key, m) => rb.set(key, val));
```

Comment:
This can override a binding that already exists.

I think we should forbid any of these effects to have binding and property modifications unless they're part of a newly created object (so and po respectively). That way there can't be a conflict.